src/game/Spells/Spell.cpp

```diff
@@ -1239,15 +1239,15 @@ void Spell::DoAllEffectOnTarget(TargetInfo *target)
     }
     else                                                    // in 1.12.1 we need explicit miss info
     {
-        if (missInfo == SPELL_MISS_MISS || missInfo == SPELL_MISS_RESIST)
+        if (missInfo == SPELL_MISS_MISS || missInfo == SPELL_MISS_RESIST || missInfo == SPELL_MISS_IMMUNE)
         {
             if (pRealUnitCaster && pRealUnitCaster != unit)
             {
                 if (!m_spellInfo->IsPositiveSpell() && (m_caster->IsVisibleForOrDetect(unit, unit, false) || m_spellInfo->HasAttribute(SPELL_ATTR_EX_IS_PICKPOCKET)))
                 {
-                    bool combat = !IsTriggeredByAura() &&
-                                  !m_spellInfo->HasAttribute(SPELL_ATTR_EX_NO_THREAT) &&
-                                  !m_spellInfo->HasAttribute(SPELL_ATTR_EX2_NO_INITIAL_THREAT);
+                    bool combat = !m_spellInfo->HasAttribute(SPELL_ATTR_EX_NO_THREAT) &&
+                                  (m_immediateHandled || !IsTriggeredByAura() &&
```

Comment:
What is going on here, why are you bypassing the No Initial Threat attribute?

Reply:
The issue here was when a delayed spell like arcane missiles would resist because the spell has no initial threat and was also triggered by an aura this would skip the combat section and not cause any aggro with the target. If the immediate portion of the channeling spell is already handled then we can ignore the initial threat attribute. That was my thinking. I tested this and it fixes the issue with arcane missiles. Other channeling spells like drain life still work. I'm not sure if this could break other spells though.

Reply:
This was the code before the arcane missiles change that made resists not start combat:

https://github.com/vmangos/core/commit/73cab4e5a2aba6cc38d5e42afae2b54220da82e7